compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -138,6 +138,13 @@ static void addTileAndDistributePasses(OpPassManager &pm) {
       createFoldAffineMinInDistributedLoopsPass());
   nestedModulePM.addPass(createCanonicalizerPass());
   nestedModulePM.addPass(createCSEPass());
+
+  // We need this pass to run before any non-dristribution loop is introduced so
+  // that some redundant `tensor.empty` ops are eliminated.
+  // TODO(dcaballe): This functionality should be implemented as part of
+  // ConvertDPS?
+  pm.nest<ModuleOp>().addPass(createEliminateEmptyTensorsPass());
```

Comment:
I just file an issue and we can reference the issue id here, see https://github.com/openxla/iree/issues/14316